nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -228,11 +228,37 @@ let
       ${lib.getExe cfg.tpm.package} \
         socket \
         --tpmstate dir="$NIX_SWTPM_DIR" \
-        --ctrl type=unixio,path="$NIX_SWTPM_DIR"/socket,terminate \
+        --server type=unixio,path="$NIX_SWTPM_DIR"/socket \
+        --ctrl type=unixio,path="$NIX_SWTPM_DIR"/socket.ctrl \
         --pid file="$NIX_SWTPM_DIR"/pid --daemon \
+        --flags not-need-init \
         --tpm2 \
         --log file="$NIX_SWTPM_DIR"/stdout,level=6
 
+      (
+        export TPM2TOOLS_TCTI=swtpm:path="$NIX_SWTPM_DIR"/socket
+        export PATH=${
+          lib.makeBinPath [
+            pkgs.tpm2-tools
+          ]
+        }:$PATH
+
+        tpm2_startup --clear
+        tpm2_startup
+
+        ${lib.optionalString (cfg.tpm.provisioning != null)
+          # Run provisinoning in a subshell not to pollute vars
+          ''
+            (
+                        export TCTI=swtpm:path="$NIX_SWTPM_DIR"/socket
```

Comment:
`export TCTI="$TPM2TOOLS_TCTI"` here? (which utilities use $TCTI vs the TPM2TOOLS_ version - is this even necessary?)

Reply:
If you were to use `tpm2-openssl`, then that would be `TPM2OPENSSL_TCTI`, everyone has its flavor. I thought providing a "generic" variable name was a good idea (that way you would re-export what you would need when you need it).

Otherwise I expect people to mostly use tpm2-tools.

Reply:
Oh, that makes sense. tpm2-pkcs11 even has its own version too. Maybe add some documentation that $TCTI is exported for this purpose in the option description.